lib/lcdbwf/R/helpers.R

```diff
@@ -493,59 +504,413 @@ nested.lapply <- function(x, subfunc, ...){
 #'       up=list(...),
 #'       down=list(...)
 #'     ),
+#'     het.vs.wt_v2=list(
+#'       res='het.vs.wt',
+#'       up=list(...),
+#'       down=list(...)
+#'     ),
 #'     ...
 #'  )
 #'
 #'
-compose_results <- function(res_list,
-                            dds_list,
+compose_results <- function(res_list=NULL,
+                            dds_list=NULL,
                             rld_list=NULL,
                             enrich_list=NULL,
-                            degpatterns_list=NULL){
+                            degpatterns_list=NULL,
+                            all_dds=NULL,
+                            all_rld=NULL,
+                            rds_file=NULL,
+                            workers=1){
+
+  if(is.null(res_list) & is.null(dds_list) & is.null(rds_file)){
+    stop('Either "res_list" & "dds_list" or "rds_file" must be specified')
+  } else if(is.null(res_list) | is.null(dds_list)){
+    message(paste('Loading objects from RDS file:', rds_file))
+
+    if(!file.exists(rds_file)){
+      stop(paste('RDS file does not exist:', rds_file))
+    }
+
+    # get res_list & dds_list (and any others) from RDS file
+    tmp <- readRDS(rds_file)
+
+    if(!any(c('res_list', 'dds_list') %in% names(tmp))){
+      stop('Object must contain "res_list" & "dds_list" elements!')
+    }
+
+    res_list <- tmp$res_list
+    dds_list <- tmp$dds_list
+
+    if('rld_list' %in% names(tmp)) rld_list <- tmp$rld_list
+    if('enrich_list' %in% names(tmp)) enrich_list <- tmp$enrich_list
+    if('degpatterns_list' %in% names(tmp)) degpatterns_list <- tmp$degpatterns_list
+    if('all_dds' %in% names(tmp)) all_dds <- tmp$all_dds
+    if('all_rld' %in% names(tmp)) all_rld <- tmp$all_rld
```

Comment:
Would be useful to add to documentation that whatever's in the RDS file overrides what's provided to this function. Actually, now that I think about it, it might be good to flip it around (whatever is provided to the function wins). That way you have the option of overriding or not. Currently you can't opt to replace what's in the RDS.

Reply:
I agree. I think it would be intuitive to have specified arguments win over what's in the RDS file. Will add that